beets/test/fixtures.py

```diff
@@ -0,0 +1,92 @@
+# This file is part of beets.
+# Copyright 2024, Arav K.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+"""
+This module provides `pytest`-based fixtures for testing Beets.
+"""
+
+from pathlib import Path
+from typing import Iterator
+
+import pytest
+from confuse import Configuration
+
+import beets
+from beets.library import Library
+
+
+@pytest.fixture(scope="session")
+def resource_dir() -> Path:
+    """
+    A fixture returning the resource directory for tests.
+    """
+
+    return Path(__file__).parents[2] / "test" / "rsrc"
+
+
+@pytest.fixture(scope="function")
+def config(
+    tmp_path: Path,
+    monkeypatch: pytest.MonkeyPatch,
+) -> Iterator[Configuration]:
+    """
+    A fixture for Beets configuration.
+    """
+
+    # 'confuse' looks at `HOME`, so we set it to a tmpdir.
+    monkeypatch.setenv("HOME", str(tmp_path))
+
+    # Initialize configuration.
+    config = beets.config
+    config.sources = []
+    config.read(user=False, defaults=True)
+
+    # Set configuration defaults.
+    config["plugins"] = []
+    config["verbose"] = 1
+    config["ui"]["color"] = False
+    config["threaded"] = False
+    config["directory"] = str(tmp_path)
+
+    # Provide the configuration to the test.
+    yield config
+
+    # Reset the global configuration state.
+    beets.config.clear()
+    beets.config._materialized = False
+
+
+@pytest.fixture(scope="function")
+def library(
+    tmp_path: Path,
+    config: Configuration,
+    monkeypatch: pytest.MonkeyPatch,
+) -> Iterator[Library]:
+    """
+    A fixture for the Beets library.
+    """
+
+    # Beets needs a location to store library contents.
+    lib_dir = tmp_path / "lib_dir"
+    lib_dir.mkdir(exist_ok=False)
```

Comment:
Instead of `tmp_path` try using [`tmp_path_factory`](https://docs.pytest.org/en/stable/how-to/tmp_path.html#the-tmp-path-factory-fixture):

```suggestion
    lib_dir = tmp_path_factory.mktemp("lib_dir")
```

Reply:
Yep, thanks!